packages/react/src/components/stepper/Stepper.tsx

```diff
@@ -0,0 +1,284 @@
+import React, { createRef, useEffect, useRef, useState } from 'react';
+
+// import core base styles
+import 'hds-core';
+import styles from './Stepper.module.scss';
+import { Step, StepProps } from './Step';
+import classNames from '../../utils/classNames';
+import { IconAngleLeft, IconAngleRight } from '../../icons';
+import { useTheme } from '../../hooks/useTheme';
+
+type Language = 'en' | 'fi' | 'sv' | string;
+
+export interface StepperCustomTheme {
+  /**
+   * A custom background color for stepper
+   */
+  '--hds-stepper-background-color'?: string;
+  /**
+   * A custom color for stepper
+   */
+  '--hds-stepper-color'?: string;
+  /**
+   * A custom width for step
+   */
+  '--hds-step-width'?: string;
+  /**
+   * A custom color for selected step label
+   */
+  '--hds-selected-step-label-color'?: string;
+  /**
+   * A custom color for not selected step label
+   */
+  '--hds-not-selected-step-label-color'?: string;
+  /**
+   * A custom color for step circle background
+   */
+  '--hds-step-background-color'?: string;
+  /**
+   * A custom color for step circle content
+   */
+  '--hds-step-content-color'?: string;
+  /**
+   * A custom color for stepper focus borders
+   */
+  '--hds-stepper-focus-border-color'?: string;
+}
+
+export type StepperProps = React.ComponentPropsWithoutRef<'button'> & {
+  /**
+   * A custom className passed to stepper
+   */
+  className?: string;
+  /**
+   * Data test id of stepper
+   */
+  dataTestId?: string;
+  /**
+   * A custom class name for step heading
+   */
+  headingClassName?: string;
+  /**
+   * The labels of the steps
+   */
+  labels: string[];
+  /**
+   * The language of the stepper
+   */
+  language?: Language;
+  /**
+   * A callback function for custom action on step click
+   */
+  onStepClick?: (event, number: number) => void;
+  /**
+   * A function for rendering a custom aria label for step's state
+   */
+  renderCustomStateAriaLabel?: (step, state) => string;
+  /**
+   * A function for rendering a custom step count label
+   */
+  renderCustomStepCountLabel?: (step, totalNumberOfSteps) => string;
+  /**
+   * A function for rendering a custom step heading
+   */
+  renderCustomStepHeading?: (step, totalNumberOfSteps, label) => string;
+  /**
+   * The selected step number. Counting starts from 1
+   */
+  selectedStep: number;
```

Comment:
Could this be optional and an index? But I'm not sure 😁 
I was also thinking that could this be replaced if every step had a unique id 🤔

Reply:
Good point. There might not be any selected states in some weird cases. And I guess it could be also an index, starting from 0

Reply:
- [ ] Refactor selectedStep as an index (starting from 0), and as optional